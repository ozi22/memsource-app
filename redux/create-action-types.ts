export const createActionTypes = (actionTypes: any = {}, duckName = 'DUCK'): any => {
  return Object.keys(actionTypes).reduce(
    (res, key) => ({
      ...res,
      [key]: `${duckName.toUpperCase()}_DUCK/${actionTypes[key].toUpperCase()}`
    }),
    {}
  );
};
