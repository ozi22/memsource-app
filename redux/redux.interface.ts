import { ThunkAction } from 'redux-thunk';

export interface IAction<Payload = any, Opt = any> {
  type: string;
  payload: Payload;
  opt?: Opt;
}

export type ThunkResult<R> = ThunkAction<R, any, undefined, any>;
