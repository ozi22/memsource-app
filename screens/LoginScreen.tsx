import React, { FC } from 'react';
import { Button, StyleSheet, Text, TextInput, View } from 'react-native';
import LoginDuck from '../redux/login/login.duck';
import UIDuck from '../redux/ui/ui.duck';
import { Controller, useForm } from 'react-hook-form';
import AuthService from '../services/auth/auth.service';
import { useDispatch, useSelector } from 'react-redux';
import Colors from '../constants/Colors';

const LoginScreen: FC = () => {
  const { control, handleSubmit, errors } = useForm();
  const isLoading: boolean = useSelector(UIDuck.isLoading);
  const dispatch = useDispatch();

  const getToken = (requestData: any) => {
    dispatch(UIDuck.setLoading(1));
    AuthService.getToken(requestData)
      .then(({ data }) => {
        dispatch(LoginDuck.addToken(data));
      })
      .catch((e) => {
        // tslint:disable-next-line:no-console
        console.error(e);
      })
      .finally(() => dispatch(UIDuck.setLoading(-1)));
  };

  const onSubmit = (data: any) => getToken(data);

  return (
    <View style={styles.container}>
      <View style={styles.inputContainer}>
        <Controller
          control={control}
          render={({ onChange, onBlur, value }) => (
            <TextInput style={styles.input} onBlur={onBlur} onChangeText={(value) => onChange(value)} value={value} />
          )}
          name="userName"
          rules={{ required: true }}
          defaultValue=""
        />
        {errors.userName && <Text style={styles.error}>This field is required.</Text>}
      </View>
      <View style={styles.inputContainer}>
        <Controller
          control={control}
          render={({ onChange, onBlur, value }) => (
            <TextInput
              style={styles.input}
              onBlur={onBlur}
              onChangeText={(value) => onChange(value)}
              value={value}
              secureTextEntry={true}
            />
          )}
          name="password"
          rules={{ required: true }}
          defaultValue=""
        />
        {errors.password && <Text style={styles.error}>This field is required.</Text>}
      </View>

      <Button color="#10b1fc" title={isLoading ? 'Submitting...' : 'Submit'} onPress={handleSubmit(onSubmit)} />
    </View>
  );
};

export default LoginScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.light.background,
    alignItems: 'center',
    justifyContent: 'center',
    padding: 15
  },
  error: {
    color: '#f13'
  },
  inputContainer: {
    maxWidth: 300,
    width: '80%',
    paddingBottom: 10
  },
  input: {
    borderColor: Colors.light.tint,
    borderStyle: 'solid',
    borderWidth: 1,
    backgroundColor: '#fff',
    height: 40,
    width: '100%',
    paddingVertical: 2,
    paddingHorizontal: 5
  }
});
