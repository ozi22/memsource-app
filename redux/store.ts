import { applyMiddleware, combineReducers, createStore } from 'redux';
import reduxThunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import UIDuck from './ui/ui.duck';
import LoginDuck from './login/login.duck';
import ProjectsDuck from './projects/projects.duck';

const combineDucks = (...ducks: any): any =>
  combineReducers(
    ducks.reduce((root: any, duck: any) => {
      return { ...root, [duck.name]: duck.reducer };
    }, {})
  );

export type AppState = ReturnType<typeof combineDucks>;

const rootReducer: any = combineDucks(UIDuck, LoginDuck, ProjectsDuck);

export const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(reduxThunk)));
