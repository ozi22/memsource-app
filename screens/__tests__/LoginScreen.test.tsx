import React from 'react';
import renderer from 'react-test-renderer';
import LoginScreen from '../LoginScreen';
import { Provider } from 'react-redux';
import { store } from '../../redux/store';

describe('LoginScreen', () => {
  it('should match snapshot', () => {
    const { toJSON } = renderer.create(
      <Provider store={store}>
        <LoginScreen />
      </Provider>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
