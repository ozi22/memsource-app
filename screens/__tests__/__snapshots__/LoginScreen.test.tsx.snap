// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f3f7f9",
      "flex": 1,
      "justifyContent": "center",
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "maxWidth": 300,
        "paddingBottom": 10,
        "width": "80%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderColor": "#10b1fc",
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 40,
          "paddingHorizontal": 5,
          "paddingVertical": 2,
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "maxWidth": 300,
        "paddingBottom": 10,
        "width": "80%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderColor": "#10b1fc",
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 40,
          "paddingHorizontal": 5,
          "paddingVertical": 2,
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#10b1fc",
            },
          ]
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;
